pkg/auth/jwks.go

```diff
@@ -0,0 +1,102 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/MicahParks/keyfunc/v3"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/spf13/pflag"
+)
+
+type JWKS struct {
+	// Endpoint to load the JWKS from.
+	//
+	// Supports schemes http, https or file.
+	Endpoint string `json:"endpoint" yaml:"endpoint"`
+
+	// How long to cache the JWKS for before reloading.
+	CacheTTL time.Duration `json:"cache_ttl" yaml:"cache_ttl"`
+
+	// Timeout for loading the JWKS.
+	Timeout time.Duration `json:"timeout" yaml:"timeout"`
+}
+
+// LoadedJWKS provides a ready to use jwt.KeyFunc for token verification.
```

Comment:
nit: can you move to the bottom of the file (I try to keep struct definitions and their methods next to each other, rather than interleaving, so here would be `JWKSConfig` struct, then its methods, then `LoadedJWKS`)